From b293e4dc27902199c4565178212c52d5ad75de9c Mon Sep 17 00:00:00 2001
From: Ken Zhu <guigenz@codeaurora.org>
Date: Fri, 5 Mar 2021 10:34:53 -0800
Subject: [PATCH] netfilter: fix source nat issue by removing the confirm check
 on nat setup

When a multicast flow lands in LAN side bridge, if bridge nf call iptables
is enabled, the flow could be sent to both WAN and some of the bridge slave
interfaces in some cases. This could lead to no source nat action because
the flow is already in a confirmed state which was just set by the multicast
packet taking bridging path.
This change removes the conntrack confirm check on nat setup.

Change-Id: I38d61a403e9484e36e2e28bf6830538921556c55
Signed-off-by: Ken Zhu <guigenz@codeaurora.org>
---
 net/netfilter/nf_nat_core.c | 4 ----
 1 file changed, 4 deletions(-)

--- a/net/netfilter/nf_nat_core.c
+++ b/net/netfilter/nf_nat_core.c
@@ -585,10 +585,6 @@ nf_nat_setup_info(struct nf_conn *ct,
 	struct net *net = nf_ct_net(ct);
 	struct nf_conntrack_tuple curr_tuple, new_tuple;
 
-	/* Can't setup nat info for confirmed ct. */
-	if (nf_ct_is_confirmed(ct))
-		return NF_ACCEPT;
-
 	WARN_ON(maniptype != NF_NAT_MANIP_SRC &&
 		maniptype != NF_NAT_MANIP_DST);
 
